contracts/Strategy.sol

```diff
@@ -53,80 +222,624 @@ contract Strategy is BaseStrategy {
+            //     - The existing choice is above the max target util
+            //   - Compare APR directly only if both are between the min and max
+            if (
+                ((utilization < lowestUtilization &&
+                    (lowestUtilization > KASHI_MAXIMUM_TARGET_UTILIZATION ||
+                        utilization < KASHI_MINIMUM_TARGET_UTILIZATION)) ||
+                    (interestPerBlock < lowestInterest &&
+                        utilization < KASHI_MAXIMUM_TARGET_UTILIZATION &&
+                        utilization > KASHI_MINIMUM_TARGET_UTILIZATION &&
+                        lowestUtilization < KASHI_MAXIMUM_TARGET_UTILIZATION &&
+                        lowestUtilization >
+                        KASHI_MINIMUM_TARGET_UTILIZATION)) &&
+                kashiFractionTotal(i) > dustThreshold &&
+                kashiPairLiquidShares(i) >= minLiquidShares
+            ) {
+                lowestInterest = interestPerBlock;
+                _lowestIndex = i;
+            }
+        }
+    }
+
+    function kashiPairUtilization(IKashiPair kashiPair, uint256 sharesToDeposit)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 totalAssetShares = kashiPair.totalAsset().elastic;
+        uint256 totalBorrowAmount = kashiPair.totalBorrow().elastic;
+        uint256 fullAssetAmount =
+            bentoSharesToWant(totalAssetShares.add(sharesToDeposit), false).add(
+                totalBorrowAmount
+            );
+
+        return
+            uint256(totalBorrowAmount).mul(KASHI_UTILIZATION_PRECISION).div(
+                fullAssetAmount
+            );
+    }
+
+    function wantToBentoShares(uint256 wantAmount, bool roundUp)
+        internal
+        view
+        returns (uint256)
+    {
+        return bentoBox.toShare(BIERC20(address(this)), wantAmount, roundUp);
+    }
+
+    function bentoSharesToWant(uint256 bentoShares, bool roundUp)
+        internal
+        view
+        returns (uint256)
+    {
+        return bentoBox.toAmount(BIERC20(address(this)), bentoShares, roundUp);
+    }
+
+    function bentoSharesToKashiFraction(
+        IKashiPair kashiPair,
+        uint256 bentoShares,
+        bool roundUp
+    ) internal view returns (uint256 _kashiFraction) {
+        // Adapted from https://github.com/sushiswap/kashi-lending/blob/b6e3521d8628a835935c94a9039cfd192044d66b/contracts/KashiPair.sol#L320-L323
+        Rebase memory totalAsset = kashiPair.totalAsset();
+        Rebase memory totalBorrow = kashiPair.totalBorrow();
+        uint256 allShare =
+            uint256(totalAsset.elastic).add(
+                wantToBentoShares(totalBorrow.elastic, roundUp)
+            );
+        _kashiFraction = allShare == 0
+            ? bentoShares
+            : bentoShares.mul(totalAsset.base).div(allShare);
+    }
+
+    function kashiFractionToBentoShares(
+        IKashiPair kashiPair,
+        uint256 _kashiFraction,
+        bool roundUp
```

Comment:
Cannot find a call where this argument is false, hence in smells to me that is not needed as an argument and instead seems to be either a contract field or hardcoded value to call Kashi, wdyt?

Reply:
👍